src/components/container/IntradayHeartRateChart/IntradayHeartRateChart.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState, useContext } from 'react'
+import { DateRangeContext } from '../../presentational/DateRangeCoordinator/DateRangeCoordinator'
+import { add, eachMinuteOfInterval, format, startOfDay } from 'date-fns'
+import { LayoutContext, LoadingIndicator } from '../../presentational'
+import { HalfDayData, FullDayData, MissingMidDayData } from "./IntradayHeartRateChart.data"
+import { ColorDefinition, resolveColor } from '../../../helpers/colors'
+import { useInitializeView } from '../../../helpers/Initialization'
+import { DeviceDataV2Namespace } from '@careevolution/mydatahelps-js'
+import { ChartThreshold, IntradayHeartRateAggregationOption, MultiSeriesLineChartOptions, combinedIntradayHeartRateDataProvider } from '../../..'
+import { IntradayHeartRateData } from '../../../helpers/heart-rate-data-providers/combined-intraday-heart-rate-providers'
+import TimeSeriesChart from '../../presentational/TimeSeriesChart'
+import { isNumber } from 'lodash'
+
+export type IntradayHeartRatePreviewState = "Default" | "CompleteDataWithThresholds" | "MissingMidDayDataThresholds" | "PartialDataWithThresholds" | "NoData";
+
+export interface IntradayHeartRateChartProps {
+    previewState?: IntradayHeartRatePreviewState,
+    dataSources: DeviceDataV2Namespace[],
+    aggregationOption: IntradayHeartRateAggregationOption,
+    aggregationIntervalMinutes: number,
+    lineColor?: ColorDefinition,
+    thresholds?: ChartThreshold[],
+    innerRef?: React.Ref<HTMLDivElement>
+}
+
+export interface IntradayDataPoint {
+    timestamp: number 
+    date: Date
+    value?: number
+    rawValue?: number
+}
+
+export default function IntradayHeartRateChart(props: IntradayHeartRateChartProps) {
+    const [loading, setLoading] = useState(true);
+    const [data, setData] = useState<IntradayDataPoint[] | undefined>(undefined);
+    const [options, setOptions] = useState<MultiSeriesLineChartOptions | undefined>(undefined);
+    const dateRangeContext = useContext(DateRangeContext);
+    const intervalStart = startOfDay(dateRangeContext?.intervalStart ?? new Date());
+    const intervalEnd = startOfDay(add(intervalStart, { days: 1 }));
```

Comment:
Minor:  I think you can drop the `startOfDay` here since `intervalStart` will already be at the start of the day.